app/models/case_contact.rb

```diff
@@ -40,18 +40,35 @@ class CaseContact < ApplicationRecord
   scope :occurred_between, ->(start_date = nil, end_date = nil) {
     where("occurred_at BETWEEN ? AND ?", start_date, end_date) if start_date.present? && end_date.present?
   }
+  scope :occurred_starting_at, ->(start_date = nil) {
+    where("occurred_at >= ?", start_date) if start_date.present?
+  }
+  scope :occurred_ending_at, ->(end_date = nil) {
+    where("occurred_at <= ?", end_date) if end_date.present?
+  }
   scope :contact_made, ->(contact_made = nil) {
-    where(contact_made: contact_made) if contact_made == true || contact_made == false
+    where(contact_made: contact_made) if /true|false/.match?(contact_made.to_s)
   }
   scope :has_transitioned, ->(has_transitioned = nil) {
-    joins(:casa_case).where(casa_cases: {transition_aged_youth: has_transitioned}) if has_transitioned == true || has_transitioned == false
+    if /true|false/.match?(has_transitioned.to_s)
```

Comment:
regex rather than boolean? why?

Reply:
I used regex because the filter coming from the request will always be a string (and the previous `if` didn't work with it)

That was the best solution I could think of at the moment

Reply:
in this case stability ~= fragility, we have been instead choosing flexibility and trusting the tests... so we need lots of tests :)

Reply:
I see, very interesting.